### ISSUE TITLE
Variable creation fails in TF-Nightly

### ISSUE THREAD

**Issue by seanpmorgan** [tensorflow/tensorflow#33348]
**System information**
- Have I written custom code (as opposed to using a stock example script provided in TensorFlow): No
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): Ubuntu 18.04
- TensorFlow installed from (source or binary): Binary
- TensorFlow version (use command below):  > 2.1.0-dev20191012
- Python version: 3.6

**Describe the current behavior**
Variable creation currently fails with:
```
  File "/usr/local/lib/python2.7/dist-packages/tensorflow_core/python/ops/variables.py", line 261, in __call__
    return cls._variable_v2_call(*args, **kwargs)
  File "/usr/local/lib/python2.7/dist-packages/tensorflow_core/python/ops/variables.py", line 255, in _variable_v2_call
    shape=shape)
  File "/usr/local/lib/python2.7/dist-packages/tensorflow_core/python/ops/variables.py", line 236, in <lambda>
    previous_getter = lambda **kws: default_variable_creator_v2(None, **kws)
  File "/usr/local/lib/python2.7/dist-packages/tensorflow_core/python/ops/variable_scope.py", line 2645, in default_variable_creator_v2
    shape=shape)
  File "/usr/local/lib/python2.7/dist-packages/tensorflow_core/python/ops/variables.py", line 263, in __call__
    return super(VariableMetaclass, cls).__call__(*args, **kwargs)
  File "/usr/local/lib/python2.7/dist-packages/tensorflow_core/python/ops/resource_variable_ops.py", line 1410, in __init__
    distribute_strategy=distribute_strategy)
  File "/usr/local/lib/python2.7/dist-packages/tensorflow_core/python/ops/resource_variable_ops.py", line 1556, in _init_from_args
    graph_mode=self._in_graph_mode)
  File "/usr/local/lib/python2.7/dist-packages/tensorflow_core/python/ops/resource_variable_ops.py", line 231, in eager_safe_variable_handle
    shape, dtype, shared_name, name, graph_mode, initial_value)
  File "/usr/local/lib/python2.7/dist-packages/tensorflow_core/python/ops/resource_variable_ops.py", line 167, in _variable_handle_from_shape_and_dtype
    handle_data.shape_and_type.append(
AttributeError: 'google.protobuf.pyext._message.RepeatedCompositeCo' object has no attribute 'append'
```
**Code to reproduce the issue**
From example: https://www.tensorflow.org/guide/variable#create_a_variable
```
import tensorflow as tf
var = tf.Variable(tf.zeros([1., 2., 3.]))
```

**Other info / logs**
I believe this is occured after aa25ad70c021968fb3a4a93ee814ca2fa699b32b 

cc @mrry

**Comment by mrry** [comment]
That's odd, because this code works in our CI. Which `protobuf` version are you using? It looks like `RepeatedCompositeContainer.append()` has been available since https://github.com/protocolbuffers/protobuf/commit/d8c2501b43c1b56e3efa74048a18f8ce06ba07fe , so we might need to bump a requirement.

**Comment by seanpmorgan** [comment]
Hmmm this is for the binary pip installed `tf-nightly` so not sure what version protobuf that is built with. Quick glance looks [like maybe 3.3.1](https://github.com/tensorflow/tensorflow/blob/master/tensorflow/tools/ci_build/protobuf/protobuf_optimized_pip.sh#L17), but I'm not overly familiar with the toolchain for TF-Core. 

As a data point... variable creation for today's MacOS nightly wheel works fine, however the linux wheel appears broken. Is there a mismatch in protobuf versions for the nightly builds?
https://colab.research.google.com/drive/11swg1klzul_ozOQZa61Wz-zoBJg25He3

cc @angerson

**Comment by mrry** [comment]
For binaries, the PIP dependency is `'protobuf >= 3.6.1'`:

https://github.com/tensorflow/tensorflow/blob/master/tensorflow/tools/pip_package/setup.py#L64

Right enough, https://github.com/protocolbuffers/protobuf/commit/d8c2501b43c1b56e3efa74048a18f8ce06ba07fe only appears from 3.8.0 onwards.

If you do `pip install -U protobuf=3.8.0` in your Linux env, does that fix the problem?

**Comment by seanpmorgan** [comment]
Yup upgrade protobuf solves the issue:
https://colab.research.google.com/drive/11swg1klzul_ozOQZa61Wz-zoBJg25He3

Can we bump the requirements in `setup.py`

**Comment by mrry** [comment]
Thanks for confirming! I just sent an PR to bump the requirement.

**Comment by Arpan2798** [comment]
When I upgrade protobuf from version 3.6.0 to any higher versions, its showing 
"ImportError: DLL load failed: The specified procedure could not be found". Earlier I was using 3.11 but that was giving me the above error but someone suggested to downgrade the version to 3.6. After downgrading its now showing this same attribute error.

**Comment by Gfqwy** [comment]
python3.6 (anaconda)
tensorflow2.1.0
![image](https://user-images.githubusercontent.com/52851180/80276101-c4dcf000-8718-11ea-9165-be33091e3157.png)
![image](https://user-images.githubusercontent.com/52851180/80276107-d0c8b200-8718-11ea-82c1-05541eac48d0.png)

That("tf.Variable(5)") used by tensorflow2.0.0 is ok, but it isn't used by tensorflow2.1.0.
what I can do some change

**Comment by Melykuti** [comment]
It might help other users if I report that I had error messages
`AttributeError: 'RepeatedCompositeFieldContainer' object has no attribute 'append'`
trying to run basic Tensorflow functionalities, like  
`model = tf.keras.models.Sequential(...)`
or
`pred = model.predict(dataset_X)`, where instead of training a model, I had specified it using `tf.constant_initializer(...)`.

This happened on a fresh install of Python 3.8 and Tensorflow 2.2, but also previously on Python (~)3.7 with Tensorflow 2.1, while it was all functioning in Tensorflow 2.0.

The solution was to upgrade `protobuf` from 3.6.1 to 3.11.3, and also to make sure that the `PYTHONPATH` was correct. My modified `~/.bashrc` had to be updated so that 
`export PYTHONPATH="/usr/local/lib64/python3.6/site-packages"`
was replaced with
`export PYTHONPATH="/usr/local/lib64/python3.8/site-packages"`
Otherwise Python found exclusively the older version of protobuf in /usr/local/lib64/python3.6/site-packages.
